_posts/2025-02-20-use-desctop-visual-studio-code.adoc

```diff
@@ -0,0 +1,108 @@
+---
+title: "Develop in local Visual Studio Code using Eclipse Che as a remote Cloud Development Environment"
+layout: post
+author: Vitaliy Gulyy
+description: >-
+   Use your local Visual Studio Code as an editor for a Che workspace
+categories: []
+keywords: ['Visual Studio Code', 'Eclipse Che editor']
+slug: /@vgulyy/use-desktop-visual-studio-code
+---
+
+== Introduction
+
+Eclipse Che provides developers a web version of the of link:https://github.com/microsoft/vscode[Visual Studio Code OSS] editor, that includes all the features of the desktop version. But for developers, the transition to cloud development can be quite unusual.
+With link:https://code.visualstudio.com/docs/editor/command-line[Visual Studio Code Command Line Interface (CLI)] you can create a tunnel that alows you connect your familiar local Visual Studio Code to your workspace.
+
+== Connecting Visual Studio Code desktop to a running workspace
+
+=== Prepare your local Visual Studio Code.
+
+The first, you need to install both link:https://marketplace.visualstudio.com/items?itemName=ms-vscode.remote-server[Remote - Tunnels] and link:https://marketplace.visualstudio.com/items?itemName=ms-vscode.remote-explorer[Remote Explorer] extensions to your local Visual Studio Code. It will allow you connect to a remote machine via a secure tunnel.
+
+=== Install Visual Studio Code CLI.
+
+Then you need to install Visual Studio Code CLI in your workspace. Open a terminal and run the following command to install Visual Studio Code CLI in the `/checode` directory.
```

Comment:
```suggestion
Then you need to install Visual Studio Code CLI in your workspace. Open a terminal and run the following command to install Visual Studio Code CLI in the `/checode` directory.
```
In the future, I'm wondering if we can pre-install this CLI and bundle it with che-code?

Reply:
I managed to build CLI from sources, but it seems we have to pre-install the code-server as well. I haven't tried to build the code-server, so I can't say much about it.